File: Week6/Day4/E2E_basics/poms/wiki_home.py
from selenium.webdriver.chrome.webdriver import WebDriver
from selenium.webdriver.common.by import By
from selenium.webdriver.remote.webelement import WebElement
from selenium.webdriver.support.select import Select


class WikiHome:

    def __init__(self, driver: WebDriver):
        """by injecting the driver into the POM we make it so the driver can access the necessary elements on the web page"""
        self.driver = driver

    def english(self):
        """this method return the English link so that selenium can interact with it"""
        element: WebElement = self.driver.find_element(By.ID, "js-link-box-en")
        return element
        # return self.driver.find_element(By.ID,"js-link-box-en")

    def spanish(self):
        """for css selector, follow this format: element[attribute='value']"""
        element: WebElement = self.driver.find_element(By.CSS_SELECTOR, "div[lang='es']")
        return element

    def italian(self):
        """you can use xpath or relative xpath"""
        # relative xpath to element: //*[@lang="it"]/a
        element: WebElement = self.driver.find_element(By.XPATH, "/html/body/div[2]/div[8]/a")
        return element

    def search_bar(self):
        return self.driver.find_element(By.ID,"searchInput")

    def language_selector(self):
        element: Select = Select(self.driver.find_element(By.ID,"searchLanguage"))
        return element

    def search_button(self):
        return self.driver.find_element(By.XPATH,"/html/body/div[3]/form/fieldset/button")

File: Week9/Day3/src/test/java/com/revature/poms/WikiHome.java
package com.revature.poms;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class WikiHome {

    private WebDriver driver;
    private By searchBar = By.id("searchInput");
    private By searchButton = By.className("pure-button");

    public WikiHome(WebDriver driver){
        this.driver = driver;
        // the page factory abstracts away the logic for interacting with web elements
        PageFactory.initElements(driver,this);
    }

    // we use FindBy to tell the page factory what selector and associated values to use to find the element we want
    @FindBy(id = "js-link-box-en")
    public WebElement english;

    @FindBy(css = "div[lang='es']")
    public WebElement spanish;

    @FindBy(xpath = "/html/body/div[2]/div[8]/a")
    public WebElement italian;

    // you can also avoid using the page factory and manually search for the elements youtself
    public void sendKeysToSearchbar(String input){
        driver.findElement(searchBar).sendKeys(input);
    }

    public void clickSearchButton(){
        driver.findElement(searchButton).click();
    }

}
